src/ansys/sound/core/sound_composer/source_broadband_noise.py

```diff
@@ -0,0 +1,334 @@
+                f"\t\t{str_values}"
+            )
+        else:
+            str_source = "Not set"
+
+        # Source control info.
+        if self.is_source_control_valid():
+            str_source_control = (
+                f"{self.source_control.control.name}\n"
+                f"\tMin: {self.source_control.control.data.min()}\n"
+                f"\tMax: {self.source_control.control.data.max()}\n"
+                f"\tDuration: "
+                f"{self.source_control.control.time_freq_support.time_frequencies.data[-1]} s"
+            )
+        else:
+            str_source_control = "Not set"
+
+        return f"Broadband noise source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlTime:
+        """Broadband noise source control.
+
+        Contains the control parameter values over time.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(source_control, SourceControlTime) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlTime``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_bbn(self) -> FieldsContainer:
+        """Broadband noise source data, as a DPF fields container.
+
+        The broadband noise source data consists of a series of spectra, each corresponding to a
+        control parameter value. Spectra can be narrowband, PSD, octave-band levels, or
+        1/3-octave-band levels.
+        """
+        return self.__source_bbn
+
+    @source_bbn.setter
+    def source_bbn(self, source_bbn: FieldsContainer):
+        """Set the broadband noise source data, from a DPF fields container."""
+        if source_bbn is not None:
+            if not isinstance(source_bbn, FieldsContainer):
+                raise PyAnsysSoundException(
+                    "Specified broadband noise source must be provided as a DPF fields container."
+                )
+
+            if len(source_bbn) < 1:
+                raise PyAnsysSoundException(
+                    "Specified broadband noise source must contain at least one spectrum."
+                )
+
+            for spectrum in source_bbn:
+                if len(spectrum.data) < 1:
+                    raise PyAnsysSoundException(
+                        "Each spectrum in the specified broadband noise source must contain at "
+                        "least one element."
+                    )
+
+            support_data = source_bbn.get_support("control_parameter_1")
+            support_properties = support_data.available_field_supported_properties()
+            support_values = support_data.field_support_by_property(support_properties[0])
+            if len(support_values) < 1:
```

Comment:
if len(support_value) != len(source_bbn) < 1: "Control data in the support associated to the FieldsContainer must contain as many elements as the number of fields in the FieldsContainer"

Reply:
Done, although I changed the message to emphasize the problem in the data (spectra vs control values) rather than in the data structure (FieldsContainer)